configure.ac

```diff
@@ -3991,6 +3991,21 @@ AS_VAR_IF(
    LIBMPDEC_LIBS="-lm \$(LIBMPDEC_A)"
    LIBMPDEC_INTERNAL="\$(LIBMPDEC_HEADERS) \$(LIBMPDEC_A)"])
 
+AS_VAR_IF([with_system_libmpdec], [yes],
+  [WITH_SAVE_ENV([
+    CPPFLAGS="$LIBMPDEC_CFLAGS $CPPFLAGS"
+    LDFLAGS="$LIBMPDEC_LIBS $LDFLAGS"
+    AC_SEARCH_LIBS(
```

Comment:
Just for curiosity's sake, why `AC_SEARCH_LIBS` rather than `AC_CHECK_LIB`?  The latter seems to be significantly more prevalent in our `configure.ac`, but I'm not aware of any significant difference between them in the case of only one library.

Reply:
`AC_SEARCH_LIBS` is the recommended macro, according to GNU Autoconf docs, but I don't think it really matters for this case. Happy to change it if you prefer `AC_CHECK_LIB`.

Reply:
Let's just keep this for now.